### PR TITLE
FEATURE: BEND-62 Onboarding endpoint

### PR DESCRIPTION
**Pull request by Kehomaxd**
- add mobility in current situation
- add current situation in "me" interface

### CODE REVIEW

**Review by davidcode10 — APPROVED** [review]

**Review by PatyVilas — DISMISSED** [review]

**Review by OvidioSantoro — APPROVED** [review]

File: src/http/actions/Auth.ts [hunk 1]
```diff
@@ -183,4 +189,55 @@ export module AuthActions {
...
+      name: dto.name,
+      username: dto.username,
+      email: dto.email,
+      password: dto.password,
+      phone: dto.phone,
+      birthdate: dto.birthdate,
+      address: dto.address,
+      selfTaught: dto.selfTaught,
+      currentSituation: dto.currentSitation,
+      profilePhoto: dto.photo,
```
Comment (octapf) [inline src/http/actions/Auth.ts:215]: dto.photos should be plural > photos , because it's an array

File: src/http/controller/Auth.ts [hunk 2]
```diff
@@ -262,8 +269,48 @@ export module AuthController {
     res: express.Response,
     next: express.NextFunction
   ) => {
-    //TODO: Pending to create all the stuff
-
+    const dto = OnBoardingDTOManager.CreateOnBoardingDTOManager(req.body);
```
Comment (octapf) [inline src/http/controller/Auth.ts:272]: call this const > onBoadingDTO

File: src/models/model.currentSituation.ts [hunk 3]
```diff
@@ -39,6 +39,14 @@ const CurrentSituationSchema = new Schema({
       message: value => `${value} is not a value allowed`,
     },
   },
+  mobility: {
+    type: String,
+    require: [true, 'Work Mobility is required'],
+    enum: {
+      values: ['Yes', 'No'],
```
Comment (octapf) [inline src/models/model.currentSituation.ts:46]: change to boolean

File: src/models/types/gen/dto/dto.onboarding.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  name: Name;
+  username: string;
+  email: string;
+  password: string;
+  phone: string;
+  birthdate: Date;
+  address: Address;
+  selfTaught: boolean;
+  currentSitation: CurrentSituation;
+  photo: Photo;
```
Comment (octapf) [inline src/models/types/gen/dto/dto.onboarding.ts:18]: change name field to photos and change type to Photo[]

**Review by octapf — CHANGES_REQUESTED** [review]

**Review by OvidioSantoro — DISMISSED** [review]

**Review by PatyVilas — APPROVED** [review]

**Review by davidcode10 — APPROVED** [review]

File: src/models/model.currentSituation.ts [hunk 5]
```diff
@@ -40,10 +40,10 @@ const CurrentSituationSchema = new Schema({
     },
   },
   mobility: {
-    type: String,
+    type: Boolean,
     require: [true, 'Work Mobility is required'],
     enum: {
-      values: ['Yes', 'No'],
+      values: [true, false],
```
Comment (octapf) [inline src/models/model.currentSituation.ts:46]: remove the enum. Look at the selftaught case:

selfTaught: {
    type: Boolean,
    required: true,
    validate: {
      validator: function (): Boolean {
        return true;
      },
      message: props => `${props.value} is not a valid option`,
    },
  },

**Review by PatyVilas — APPROVED** [review]

**Review by OvidioSantoro — APPROVED** [review]

**Review by octapf — APPROVED** [review]

**Review by PatyVilas — APPROVED** [review]

**Pull request merged**